book/appendix/virtualenv.md

```diff
@@ -96,12 +96,10 @@ $HOME/wallaroo-tutorial/wallaroo/machida/build/machida --application-module cels
   --ponynoblock
 ```
 
-Please be aware that you'll still need to setup a source and sink as well. We include a toolkit named Giles to assist in this. To learn more you can look at [Giles Receiver](/book/wallaroo-tools/giles-receiver.md) and [Giles Sender](/book/wallaroo-tools/giles-sender.md) for more detailed instructions on building and running Giles.
-
-To run a sink for our example with Giles, run
+To run a sink for our example with `data_receiver`, run
 
 ```bash
-$HOME/wallaroo-tutorial/wallaroo/giles/receiver/receiver --listen 127.0.0.1:5555 --ponythreads=1 --ponynoblock
+$HOME/wallaroo-tutorial/wallaroo/utils/data_receiver/data_receiver --listen 127.0.0.1:5555 --ponythreads=1 --ponynoblock
```

Comment:
for `data_receiver` to be an in-place replacement for giles receiver as a sink, it needs to be used with the `--framed` argument. Otherwise it operates like netcat (e.g. read all available bytes, print all read bytes, goto 0), and this can look awkward on the screen.

To summarise:

|input type|input mode| output destination| output mode|
| - | - | - | - |
| framed | `--framed` | SDTOUT | write (no `--no-write`) |
| framed | streaming (no `--framed`) | FILE (via `>` etc) | write (no `--no-write`) |
| framed | either | NULL (max performance) | `--no-write` |
| steaming | streaming (no `--framed`) | STDOUT | write |
| streaming | streaming | FILE | write |
| streaming | streaming | NULL (max performance) | `--no-write` |